tests/cxx/implicit_ctor.cc

```diff
@@ -58,6 +62,55 @@ IndirectWithImplicitCtor indirect
     // IWYU: IndirectWithImplicitCtor is...*implicit_ctor-i2.h
     = 1;
 
+// IWYU: NoAutocastCtor is...*implicit_ctor-i2.h
+using Providing = NoAutocastCtor;
+// IWYU: NoTrivialCtorDtor is...*implicit_ctor-i2.h
+using NoTrivialCtorDtorProvidingAlias = NoTrivialCtorDtor;
+
+// IWYU: NoAutocastCtor needs a declaration
+void FnTakingDirectly(NoAutocastCtor);
+void FnTakingByProvidingAlias(Providing);
+void FnTakingByNonProvidingAlias(NonProviding);
+
+template <typename T>
+void TplFn(T t);
+
+// IWYU: NoAutocastCtor needs a declaration
+void TestNonAutocastConstruction(const NoAutocastCtor& par) {
+  // IWYU: NoAutocastCtor is...*implicit_ctor-i2.h
+  FnTakingDirectly(par);
+  FnTakingByProvidingAlias(par);
+  // IWYU: NoAutocastCtor is...*implicit_ctor-i2.h
+  FnTakingByNonProvidingAlias(par);
```

Comment:
In fact, I'm not sure that cases of function argument initialization should not be considered as "autocast".

Reply:
Can you elaborate? I don't fully understand what you're saying above.